ui/app/src/Views/AlertList/Views.elm

```diff
@@ -3,42 +3,139 @@ module Views.AlertList.Views exposing (view)
+                    , li [ class "nav-item ml-auto " ]
+                        [ label [ class "mt-1 custom-control custom-checkbox" ]
+                            [ input
+                                [ type_ "checkbox"
+                                , class "custom-control-input"
+                                , checked (Maybe.withDefault False filter.showSilenced)
+                                , onCheck (ToggleSilenced >> MsgForAlertList)
+                                ]
+                                []
+                            , span [ class "custom-control-indicator" ] []
+                            , span [ class "custom-control-description" ] [ text "Show Silenced" ]
+                            ]
+                        ]
+                    ]
+                ]
+            , div [ class "card-block" ]
+                [ case tab of
+                    FilterTab ->
+                        Html.map (MsgForFilterBar >> MsgForAlertList) (FilterBar.view filterBar)
+
+                    GroupTab ->
+                        Html.map (MsgForGroupBar >> MsgForAlertList) (GroupBar.view groupBar)
+                ]
+            ]
         , case alerts of
             Success groups ->
-                alertList groups filter
+                let
+                    g =
+                        silenced filter.showSilenced groups
+                            |> Utils.List.groupBy
+                                (\alert ->
+                                    List.filterMap
+                                        (\( key, value ) ->
+                                            -- Find the correct keys and return
+                                            -- their values
+                                            if List.member key groupBar.fields then
+                                                Just ( key, value )
+                                            else
+                                                Nothing
+                                        )
+                                        alert.labels
+                                )
+                in
+                    div []
+                        (List.filterMap
+                            (\k ->
+                                let
+                                    alerts =
+                                        (Dict.get k g)
+                                in
+                                    case alerts of
+                                        Just alts ->
+                                            Just (alertList k alts filter)
+
+                                        Nothing ->
+                                            Nothing
+                            )
+                            (Dict.keys g
+                                |> List.partition ((/=) [])
+                                |> uncurry (++)
+                            )
+                        )
```

Comment:
There's a lot going on here, would be nice to break it down into named functions so it's a bit easier to read and reason about.